src/framework/storage/FileSystemGraphObjectStore/FileSystemGraphObjectStore.ts

```diff
@@ -0,0 +1,138 @@
+import pMap from 'p-map';
+import { Sema } from 'async-sema';
+
+import { Entity, Relationship } from '../../types';
+import { GraphObjectFilter, GraphObjectIteratee } from '../../execution/types';
+
+import { flushDataToDisk } from './flushDataToDisk';
+import { BucketMap } from './BucketMap';
+
+import {
+  iterateEntityTypeIndex,
+  iterateRelationshipTypeIndex,
+} from './indices';
+
+export const GRAPH_OBJECT_BUFFER_THRESHOLD = 500; // arbitrarily selected, subject to tuning
+
+interface FileSystemGraphObjectStoreInput {
+  cacheDirectory?: string;
+}
+
+export class FileSystemGraphObjectStore {
+  readonly cacheDirectory?: string;
```

Comment:
Since this is optional, I now think the name could be `storageDirectoryPathPrefix`.